XenAdmin/Dialogs/VIFDialog.cs

```diff
@@ -274,7 +274,7 @@ public VIF NewVif()
         /// <returns></returns>
         public Proxy_VIF GetNewSettings()
         {
-            Proxy_VIF proxyVIF = new Proxy_VIF();
+            Proxy_VIF proxyVIF = ExistingVif != null ? ExistingVif.ToProxy() : new Proxy_VIF();

```

Comment:
Is line 282 (proxyVIF.VM = ExistingVif.VM.opaque_ref) still needed if you have this? I haven't checked, but it might not be.